File: 20210907/levelOrder.go
package _0210907

// LevelOrder 给你一个二叉树，请你返回其按 层序遍历 得到的节点值。
func LevelOrder(root *TreeNode) [][]int {
	res := [][]int{}
	if root == nil {
		return res
	}
	queue := []*TreeNode{root}
	for i := 0; len(queue) > 0; i++ {
		res = append(res, []int{})
		p := []*TreeNode{}
		for j := 0; j < len(queue); j++ {
			res[i] = append(res[i], queue[j].Val)
			if queue[j].Left != nil {
				p = append(p, queue[j].Left)
			}
			if queue[j].Right != nil {
				p = append(p, queue[j].Right)
			}
		}
		queue = p
	}
	return res
}

File: 剑指offer/easy/20210914/levelOrder.cpp
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
   vector<vector<int>> res;
   if(root== nullptr){
       return res;
   }
   queue<TreeNode*> q;
   q.push(root);
   while(!q.empty()){
       int n=q.size();
       vector<int> ans;
       for(int i=0;i<n;i++){
           TreeNode *node=q.front();
           q.pop();
           ans.push_back(node->val);
           if(node->left!= nullptr){
               q.push(node->left);
           }
           if(node->right!= nullptr){
               q.push(node->right);
           }
       }
       res.push_back(ans);
   }
   return res;
}
};
